tests/test_dayview.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+
+from app.routers.dayview import Event
+
+
+@pytest.fixture
+def event():
+    event_id = 123
+    color = 'red'
+    content = 'nothing'
+    start = "03/2/2021 4:05"
+    end = "03/2/2021 4:20"
+    event = Event(id=event_id, color=color,
+                  content=content, start_datetime=start,
+                  end_datetime=end)
+    return event
+
+
+def test_new_event(event):
+    assert event.id == 123
+    assert event.color == 'red'
+    assert event.content == 'nothing'
+    assert event.total_time == '04:05 - 04:20'
+    assert event.grid_position == '22 / 23'
+    assert event.length == 15
+
+
+def test_dayview_html(client, event):
+    events = [{"id": event.id, "color": event.color,
+               "content": event.content,
+               "start_datetime": "3/2/2021 04:05",
+               "end_datetime": "3/2/2021 04:20",
+               }]
+    day = {"year": 2021, "month": 2, "day": 3, "events": events}
+    response = client.post("/dayview", json=day)
+    res = response.content.decode("utf-8")
+    assert 'grid-row: 22 / 23;' in res
+    assert '<div id="event123"' in res
```

Comment:
It's probably better to use BeautifulSoup to find the `id` of the `div`.

There might be strange end-cases where the attributes of the elements won't be preserved, and you'll get something like `<div class="meow" id="event123">`, which, in the current form, will break the test function.